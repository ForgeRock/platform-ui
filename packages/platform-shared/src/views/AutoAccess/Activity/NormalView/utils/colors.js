/**
 * Copyright (c) 2023 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

export const colorsHex = ['#2ed47a', '#ffb946', '#109cf1', '#f7685b', '#61e2e2'];
export const colorsReadable = ['green', 'yellow', 'blue', 'red', 'cyan'];
