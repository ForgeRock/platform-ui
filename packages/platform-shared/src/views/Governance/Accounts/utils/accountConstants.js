/**
 * Copyright (c) 2025 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

const ACCOUNT_TYPES = {
  DEFAULT: 'default',
  MACHINE: 'machine',
};

const ACCOUNT_SUBTYPES = [
  {
    value: 'oauthToken', text: 'OAuth Token', icon: 'key', iconColor: 'blue',
  },
  {
    value: 'serviceAccount', text: 'Service Account', icon: 'badge', iconColor: 'purple',
  },
  {
    value: 'rpa', text: 'RPA', icon: 'precision_manufacturing', iconColor: 'orange',
  },
  {
    value: 'bot', text: 'BOT', icon: 'smart_toy', iconColor: 'green',
  },
  {
    value: 'accessToken', text: 'Access Token', icon: 'vpn_key', iconColor: 'brown',
  },
  {
    value: 'secret', text: 'Secret', icon: 'lock', iconColor: 'red',
  },
  {
    value: 'apiKey', text: 'API Key', icon: 'code', iconColor: 'blue',
  },
  {
    value: 'personalAccessToken', text: 'Personal Access Token', icon: 'person', iconColor: 'orange',
  },
  {
    value: 'managedAccount', text: 'Managed Account', icon: 'manage_accounts', iconColor: 'blue',
  },
  {
    value: 'teamToken', text: 'Team Token', icon: 'group', iconColor: 'teal',
  },
  {
    value: 'organizationToken', text: 'Organization Token', icon: 'domain', iconColor: 'gray',
  },
  {
    value: 'botToken', text: 'BOT Token', icon: 'memory', iconColor: 'green',
  },
  {
    value: 'webhookKey', text: 'Webhook Key', icon: 'link', iconColor: 'gray',
  },
  {
    value: 'integrationToken', text: 'Integration Token', icon: 'merge_type', iconColor: 'purple',
  },
  {
    value: 'servicePrincipal', text: 'Service Principal', icon: 'verified_user', iconColor: 'yellow',
  },
  {
    value: 'userManagedIdentity', text: 'User Managed Identity', icon: 'supervised_user_circle', iconColor: 'red',
  },
  {
    value: 'cognitoToken', text: 'Cognito Token', icon: 'login', iconColor: 'teal',
  },
  {
    value: 'lambdaAuthorizer', text: 'Lambda Authorizer', icon: 'flash_on', iconColor: 'yellow',
  },
];

export default {
  ACCOUNT_TYPES,
  ACCOUNT_SUBTYPES,
};
