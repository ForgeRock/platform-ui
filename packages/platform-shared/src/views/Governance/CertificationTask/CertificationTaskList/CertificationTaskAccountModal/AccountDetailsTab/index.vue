<!-- Copyright (c) 2023 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <div class="p-4">
    <dl class="row">
      <dt class="col-lg-4">
        {{ $t('governance.certificationTask.lineItemDetailsModal.accountDetailsTab.ownerLabel') }}
      </dt>
      <dd class="col-lg-8 mb-4">
        <BMedia
          class="align-items-center"
          no-body>
          <BImg
            class="mr-3 rounded-circle"
            height="36"
            width="36"
            :alt="details.displayName"
            :src="details.profileImage || require('@forgerock/platform-shared/src/assets/images/avatar.png')"
            fluid />
          <div class="media-body">
            <h5
              class="m-0"
              data-testid="displayName">
              {{ details.displayName }}
            </h5>
            <small
              class="text-muted"
              data-testid="userPrincipalName">
              {{ details.userPrincipalName }}
            </small>
          </div>
        </BMedia>
      </dd>
      <dt class="col-lg-4">
        {{ $t('governance.certificationTask.lineItemDetailsModal.accountDetailsTab.lastDecisionLabel') }}
      </dt>
      <dd
        class="col-lg-8 mb-4"
        data-testid="lastDecision">
        {{ account.decision || blankValueIndicator }}
      </dd>
      <dt class="col-lg-4">
        {{ $t('governance.certificationTask.lineItemDetailsModal.accountDetailsTab.lastCertifiedLabel') }}
      </dt>
      <dd
        class="col-lg-8 mb-4"
        data-testid="decisionDate">
        {{ formatDate(account.decisionDate) || blankValueIndicator }}
      </dd>
      <dt class="col-lg-4">
        {{ $t('governance.certificationTask.lineItemDetailsModal.accountDetailsTab.lastCertifiedByLabel') }}
      </dt>
      <dd
        class="col-lg-8 mb-4"
        data-testid="decisionBy">
        {{ account.decisionBy && account.decisionBy.userName || blankValueIndicator }}
      </dd>
      <dt class="col-lg-4">
        {{ $t('governance.certificationTask.lineItemDetailsModal.accountDetailsTab.provisioningMethodLabel') }}
      </dt>
      <dd class="col-lg-8 mb-4">
        {{ account.grantType || blankValueIndicator }}
      </dd>
    </dl>
  </div>
</template>

<script>
import {
  BImg,
  BMedia,
} from 'bootstrap-vue';
import { blankValueIndicator } from '@forgerock/platform-shared/src/utils/governance/constants';
import dayjs from 'dayjs';

export default {
  name: 'AccountDetailsTab',
  components: {
    BImg,
    BMedia,
  },
  props: {
    account: {
      type: Object,
      required: true,
    },
  },
  data() {
    return {
      blankValueIndicator,
      details: this.account.account,
    };
  },
  methods: {
    formatDate(date) {
      return dayjs(date).format('MMMM D, YYYY h:mm A');
    },
  },
};
</script>
