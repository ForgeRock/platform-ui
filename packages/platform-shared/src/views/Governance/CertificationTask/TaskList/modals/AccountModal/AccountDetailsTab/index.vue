<!-- Copyright (c) 2023-2024 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <div class="p-4">
    <dl class="row">
      <dt class="col-lg-4">
        {{ $t('governance.certificationTask.lineItemDetailsModal.accountDetailsTab.ownerLabel') }}
      </dt>
      <dd class="col-lg-8 mb-4">
        <BMedia
          class="align-items-center"
          no-body>
          <BImg
            class="mr-3 rounded-circle"
            height="36"
            width="36"
            :alt="details.displayName"
            :src="details.profileImage || require('@forgerock/platform-shared/src/assets/images/avatar.png')"
            fluid />
          <div class="media-body">
            <h5
              class="m-0"
              data-testid="displayName">
              {{ details.displayName }}
            </h5>
            <small
              class="text-muted"
              data-testid="userPrincipalName">
              {{ details.userPrincipalName }}
            </small>
          </div>
        </BMedia>
      </dd>
      <dt class="col-lg-4">
        {{ $t('governance.certificationTask.lineItemDetailsModal.accountDetailsTab.lastDecisionLabel') }}
      </dt>
      <dd
        class="col-lg-8 mb-4"
        data-testid="lastDecision">
        {{ decision }}
      </dd>
      <dt class="col-lg-4">
        {{ $t('governance.certificationTask.lineItemDetailsModal.accountDetailsTab.lastCertifiedLabel') }}
      </dt>
      <dd
        class="col-lg-8 mb-4"
        data-testid="decisionDate">
        {{ decisionDate }}
      </dd>
      <dt class="col-lg-4">
        {{ $t('governance.certificationTask.lineItemDetailsModal.accountDetailsTab.lastCertifiedByLabel') }}
      </dt>
      <dd
        class="col-lg-8 mb-4"
        data-testid="decisionBy">
        {{ decisionBy }}
      </dd>
      <dt class="col-lg-4">
        {{ $t('governance.certificationTask.lineItemDetailsModal.accountDetailsTab.provisioningMethodLabel') }}
      </dt>
      <dd class="col-lg-8 mb-4">
        {{ grantType }}
      </dd>
    </dl>
  </div>
</template>

<script>
import {
  BImg,
  BMedia,
} from 'bootstrap-vue';
import { blankValueIndicator } from '@forgerock/platform-shared/src/utils/governance/constants';
import dayjs from 'dayjs';

export default {
  name: 'AccountDetailsTab',
  components: {
    BImg,
    BMedia,
  },
  props: {
    grant: {
      type: Object,
      required: true,
    },
  },
  data() {
    return {
      blankValueIndicator,
      details: this.grant?.account,
    };
  },
  computed: {
    decision() {
      return this.grant?.item?.decision?.certification?.decision || blankValueIndicator;
    },
    decisionDate() {
      return this.formatDate(this.grant?.item?.decision?.certification?.decisionDate);
    },
    decisionBy() {
      const user = this.grant?.item?.decision?.certification?.decisionBy;
      return user ? user.userName : blankValueIndicator;
    },
    grantType() {
      return this.grant?.grantTypes || blankValueIndicator;
    },
  },
  methods: {
    formatDate(date) {
      if (date) return dayjs(date).format('MMMM D, YYYY h:mm A');
      return blankValueIndicator;
    },
  },
};
</script>
