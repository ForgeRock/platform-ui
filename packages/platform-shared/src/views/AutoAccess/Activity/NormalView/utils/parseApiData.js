/**
 * Copyright (c) 2023 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { colorsHex, colorsReadable } from './colors';

export default function parseApiData(response) {
  const categories = {};
  const categoryHasOther = {};
  const categoryNames = [...new Set(response.result.map((item) => item.category))];
  categoryNames.forEach((category) => {
    let categoryValue = {};
    let otherCount = 0;
    let otherCategoryData = {};
    categories[category] = [];
    categoryHasOther[category] = false;

    /**
     * This reorganizing of the API data is needed since the data provided is formatted in a way needed by the UI.
     *
     * The data is flat and need to be organized into a usable object structure.
     * It needs to be manipulated by merging values of 'other', and null together into a single field.
     * The field of 'Others' also need to be converted to have the label 'other' and be sorted last
     */
    response.result.forEach((item) => {
      if (item.category === category) {
        // 'other' or null
        if (item.value === 'other' || !item.value) {
          otherCount += parseInt(item.count, 10);
          if (!categoryHasOther[category]) {
            otherCategoryData = {
              label: 'unknown',
              value: otherCount,
              valueFormatted: otherCount,
            };
            categoryHasOther[category] = true;
          } else {
            otherCategoryData.value = otherCount;
            otherCategoryData.valueFormatted = otherCount;
          }
        } else {
          // Handle all normal categories, except 'Others' which gets renames to 'other'
          categoryValue = {
            label: item.value === 'Others' ? 'other' : item.value,
            value: parseInt(item.count, 10),
            valueFormatted: item.count,
          };
          categories[item.category].push(categoryValue);
        }
      }
    });

    // We have found 'other' or null and need to add the manipulated field to the main array
    if (categoryHasOther[category]) {
      categories[category].push(otherCategoryData);
    }

    // Sort categories based on numerical 'value', but leave 'other' sorted last
    Object.keys(categories).forEach((cat) => {
      categories[cat].sort((a, b) => {
        const aVal = parseInt(a.value, 10);
        const bVal = parseInt(b.value, 10);
        if (a.label === 'other') { return 1; }
        if (b.label === 'other') { return -1; }
        if (aVal < bVal) { return 1; }
        if (aVal > bVal) { return -1; }
        return 0;
      });

      // Set the color values - hex for the Donut Chart, readable for the data table
      let colorIndex = 0;
      categories[cat].forEach((item) => {
        item.color = colorsHex[colorIndex] || colorsHex[4];
        item.colorReadable = colorsReadable[colorIndex] || colorsReadable[4];
        colorIndex += 1;
      });
    });
  });

  return categories;
}
