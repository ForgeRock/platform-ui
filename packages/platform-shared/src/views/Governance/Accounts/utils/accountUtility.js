/**
 * Copyright (c) 2025 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { get } from 'lodash';
import { blankValueIndicator } from '@forgerock/platform-shared/src/utils/governance/constants';
import accountConstants from './accountConstants';

/**
 * Get the badge variant for various account types
 * @returns badge variant
 */
export function getAccountTypeVariant(accountType) {
  const type = accountType.toLowerCase();
  if (type === accountConstants.ACCOUNT_TYPES.DEFAULT) {
    return 'light';
  }
  if (type === accountConstants.ACCOUNT_TYPES.MACHINE) {
    return 'warning';
  }
  return 'light';
}

/**
 * Get the display name from an account
 * @param {object} account Account object
 * @returns Display name
 */
export function getAccountDisplayName(account) {
  return get(account, 'descriptor.idx./account.displayName') || account?.__NAME__ || blankValueIndicator;
}
