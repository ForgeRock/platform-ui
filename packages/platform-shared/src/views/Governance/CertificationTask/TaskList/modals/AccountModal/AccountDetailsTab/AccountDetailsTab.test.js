/**
 * Copyright (c) 2023-2024 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { shallowMount } from '@vue/test-utils';
import { findByTestId } from '@forgerock/platform-shared/src/utils/testHelpers';
import dayjs from 'dayjs';
import AccountDetailsTab from './index';

const account = {
  id: 'test',
  account: {
    displayName: 'Franklin Horne',
    userPrincipalName: 'fhorne@jstilton1973unfinishedlife.onmicrosoft.com',
  },
  accountEnabled: true,
  accountType: 'type',
  item: {
    decision: {
      certification: {
        decision: 'certify',
        decisionBy: {
          givenName: 'Foo',
          id: 'managed/user/1',
          mail: 'foo@test.com',
          sn: 'Test',
          userName: 'FooTest',
        },
        decisionDate: '2023-02-28T15:12:25+00:00',
      },
    },
  },
  displayName: 'Franklin Horne',
  givenName: 'Franklin',
  linkQualifier: 'default',
  mail: 'franklin.horne@usercreator.gen',
  mailNickname: 'fhorne',
  manager: null,
  metadata: {},
  otherMails: [],
  proxyAddresses: [],
  surname: 'Horne',
  userPrincipalName: 'fhorne@jstilton1973unfinishedlife.onmicrosoft.com',
  userType: 'Member',
};

const validateFieldByText = (wrapper, field, value) => {
  const fieldRef = findByTestId(wrapper, field);

  expect(fieldRef.text()).toEqual(value);
};

describe('AccountDetailsTab', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(AccountDetailsTab, {
      global: {
        mocks: {
          $t: (t) => t,
        },
        renderStubDefaultSlot: true,
      },
      props: {
        grant: account,
      },
    });
  });

  it('should show all account details properties', () => {
    validateFieldByText(wrapper, 'displayName', account.account.displayName);
    validateFieldByText(wrapper, 'userPrincipalName', account.account.userPrincipalName);
    validateFieldByText(wrapper, 'lastDecision', account.item.decision.certification.decision);
    validateFieldByText(wrapper, 'decisionDate', dayjs(account.item.decision.certification.decisionDate).format('MMMM D, YYYY h:mm A'));
    validateFieldByText(wrapper, 'decisionBy', account.item.decision.certification.decisionBy.userName);
  });
});
