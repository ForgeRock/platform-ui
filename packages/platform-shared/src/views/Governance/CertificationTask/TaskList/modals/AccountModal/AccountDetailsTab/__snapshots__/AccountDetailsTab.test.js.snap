// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDetailsTab should show all account details properties: console-warn-snapshot 1`] = `
[
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) Vue 3's render function API has changed. You can opt-in to the new API with:

  configureCompat({ RENDER_FUNCTION: false })

  (This can also be done per-component via the "compatConfig" option.)
  Details: https://v3-migration.vuejs.org/breaking-changes/render-function-api.html",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation GLOBAL_EXTEND) (2)",
    "
",
    " at <AccountDetailsTab",
    "account=",
    {
      "account": {
        "displayName": "Franklin Horne",
        "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      },
      "accountEnabled": true,
      "accountType": "type",
      "decision": "certify",
      "decisionBy": {
        "givenName": "Foo",
        "id": "managed/user/1",
        "mail": "foo@test.com",
        "sn": "Test",
        "userName": "FooTest",
      },
      "decisionDate": "2023-02-28T15:12:25+00:00",
      "displayName": "Franklin Horne",
      "givenName": "Franklin",
      "id": "test",
      "linkQualifier": "default",
      "mail": "franklin.horne@usercreator.gen",
      "mailNickname": "fhorne",
      "manager": null,
      "metadata": {},
      "otherMails": [],
      "proxyAddresses": [],
      "surname": "Horne",
      "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      "userType": "Member",
    },
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "blank" with v-bind value \`false\` will render blank="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BImg",
    "class="mr-3 rounded-circle"",
    "height="36"",
    "width="36"",
    " ...",
    ">",
    "
",
    " at <BMedia",
    "class="align-items-center"",
    "no-body=""",
    ">",
    "
",
    " at <AccountDetailsTab",
    "account=",
    {
      "account": {
        "displayName": "Franklin Horne",
        "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      },
      "accountEnabled": true,
      "accountType": "type",
      "decision": "certify",
      "decisionBy": {
        "givenName": "Foo",
        "id": "managed/user/1",
        "mail": "foo@test.com",
        "sn": "Test",
        "userName": "FooTest",
      },
      "decisionDate": "2023-02-28T15:12:25+00:00",
      "displayName": "Franklin Horne",
      "givenName": "Franklin",
      "id": "test",
      "linkQualifier": "default",
      "mail": "franklin.horne@usercreator.gen",
      "mailNickname": "fhorne",
      "manager": null,
      "metadata": {},
      "otherMails": [],
      "proxyAddresses": [],
      "surname": "Horne",
      "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      "userType": "Member",
    },
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "block" with v-bind value \`false\` will render block="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BImg",
    "class="mr-3 rounded-circle"",
    "height="36"",
    "width="36"",
    " ...",
    ">",
    "
",
    " at <BMedia",
    "class="align-items-center"",
    "no-body=""",
    ">",
    "
",
    " at <AccountDetailsTab",
    "account=",
    {
      "account": {
        "displayName": "Franklin Horne",
        "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      },
      "accountEnabled": true,
      "accountType": "type",
      "decision": "certify",
      "decisionBy": {
        "givenName": "Foo",
        "id": "managed/user/1",
        "mail": "foo@test.com",
        "sn": "Test",
        "userName": "FooTest",
      },
      "decisionDate": "2023-02-28T15:12:25+00:00",
      "displayName": "Franklin Horne",
      "givenName": "Franklin",
      "id": "test",
      "linkQualifier": "default",
      "mail": "franklin.horne@usercreator.gen",
      "mailNickname": "fhorne",
      "manager": null,
      "metadata": {},
      "otherMails": [],
      "proxyAddresses": [],
      "surname": "Horne",
      "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      "userType": "Member",
    },
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "center" with v-bind value \`false\` will render center="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BImg",
    "class="mr-3 rounded-circle"",
    "height="36"",
    "width="36"",
    " ...",
    ">",
    "
",
    " at <BMedia",
    "class="align-items-center"",
    "no-body=""",
    ">",
    "
",
    " at <AccountDetailsTab",
    "account=",
    {
      "account": {
        "displayName": "Franklin Horne",
        "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      },
      "accountEnabled": true,
      "accountType": "type",
      "decision": "certify",
      "decisionBy": {
        "givenName": "Foo",
        "id": "managed/user/1",
        "mail": "foo@test.com",
        "sn": "Test",
        "userName": "FooTest",
      },
      "decisionDate": "2023-02-28T15:12:25+00:00",
      "displayName": "Franklin Horne",
      "givenName": "Franklin",
      "id": "test",
      "linkQualifier": "default",
      "mail": "franklin.horne@usercreator.gen",
      "mailNickname": "fhorne",
      "manager": null,
      "metadata": {},
      "otherMails": [],
      "proxyAddresses": [],
      "surname": "Horne",
      "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      "userType": "Member",
    },
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "fluidGrow" with v-bind value \`false\` will render fluidGrow="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BImg",
    "class="mr-3 rounded-circle"",
    "height="36"",
    "width="36"",
    " ...",
    ">",
    "
",
    " at <BMedia",
    "class="align-items-center"",
    "no-body=""",
    ">",
    "
",
    " at <AccountDetailsTab",
    "account=",
    {
      "account": {
        "displayName": "Franklin Horne",
        "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      },
      "accountEnabled": true,
      "accountType": "type",
      "decision": "certify",
      "decisionBy": {
        "givenName": "Foo",
        "id": "managed/user/1",
        "mail": "foo@test.com",
        "sn": "Test",
        "userName": "FooTest",
      },
      "decisionDate": "2023-02-28T15:12:25+00:00",
      "displayName": "Franklin Horne",
      "givenName": "Franklin",
      "id": "test",
      "linkQualifier": "default",
      "mail": "franklin.horne@usercreator.gen",
      "mailNickname": "fhorne",
      "manager": null,
      "metadata": {},
      "otherMails": [],
      "proxyAddresses": [],
      "surname": "Horne",
      "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      "userType": "Member",
    },
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "left" with v-bind value \`false\` will render left="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BImg",
    "class="mr-3 rounded-circle"",
    "height="36"",
    "width="36"",
    " ...",
    ">",
    "
",
    " at <BMedia",
    "class="align-items-center"",
    "no-body=""",
    ">",
    "
",
    " at <AccountDetailsTab",
    "account=",
    {
      "account": {
        "displayName": "Franklin Horne",
        "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      },
      "accountEnabled": true,
      "accountType": "type",
      "decision": "certify",
      "decisionBy": {
        "givenName": "Foo",
        "id": "managed/user/1",
        "mail": "foo@test.com",
        "sn": "Test",
        "userName": "FooTest",
      },
      "decisionDate": "2023-02-28T15:12:25+00:00",
      "displayName": "Franklin Horne",
      "givenName": "Franklin",
      "id": "test",
      "linkQualifier": "default",
      "mail": "franklin.horne@usercreator.gen",
      "mailNickname": "fhorne",
      "manager": null,
      "metadata": {},
      "otherMails": [],
      "proxyAddresses": [],
      "surname": "Horne",
      "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      "userType": "Member",
    },
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "right" with v-bind value \`false\` will render right="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BImg",
    "class="mr-3 rounded-circle"",
    "height="36"",
    "width="36"",
    " ...",
    ">",
    "
",
    " at <BMedia",
    "class="align-items-center"",
    "no-body=""",
    ">",
    "
",
    " at <AccountDetailsTab",
    "account=",
    {
      "account": {
        "displayName": "Franklin Horne",
        "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      },
      "accountEnabled": true,
      "accountType": "type",
      "decision": "certify",
      "decisionBy": {
        "givenName": "Foo",
        "id": "managed/user/1",
        "mail": "foo@test.com",
        "sn": "Test",
        "userName": "FooTest",
      },
      "decisionDate": "2023-02-28T15:12:25+00:00",
      "displayName": "Franklin Horne",
      "givenName": "Franklin",
      "id": "test",
      "linkQualifier": "default",
      "mail": "franklin.horne@usercreator.gen",
      "mailNickname": "fhorne",
      "manager": null,
      "metadata": {},
      "otherMails": [],
      "proxyAddresses": [],
      "surname": "Horne",
      "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      "userType": "Member",
    },
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "rounded" with v-bind value \`false\` will render rounded="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BImg",
    "class="mr-3 rounded-circle"",
    "height="36"",
    "width="36"",
    " ...",
    ">",
    "
",
    " at <BMedia",
    "class="align-items-center"",
    "no-body=""",
    ">",
    "
",
    " at <AccountDetailsTab",
    "account=",
    {
      "account": {
        "displayName": "Franklin Horne",
        "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      },
      "accountEnabled": true,
      "accountType": "type",
      "decision": "certify",
      "decisionBy": {
        "givenName": "Foo",
        "id": "managed/user/1",
        "mail": "foo@test.com",
        "sn": "Test",
        "userName": "FooTest",
      },
      "decisionDate": "2023-02-28T15:12:25+00:00",
      "displayName": "Franklin Horne",
      "givenName": "Franklin",
      "id": "test",
      "linkQualifier": "default",
      "mail": "franklin.horne@usercreator.gen",
      "mailNickname": "fhorne",
      "manager": null,
      "metadata": {},
      "otherMails": [],
      "proxyAddresses": [],
      "surname": "Horne",
      "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      "userType": "Member",
    },
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "thumbnail" with v-bind value \`false\` will render thumbnail="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BImg",
    "class="mr-3 rounded-circle"",
    "height="36"",
    "width="36"",
    " ...",
    ">",
    "
",
    " at <BMedia",
    "class="align-items-center"",
    "no-body=""",
    ">",
    "
",
    " at <AccountDetailsTab",
    "account=",
    {
      "account": {
        "displayName": "Franklin Horne",
        "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      },
      "accountEnabled": true,
      "accountType": "type",
      "decision": "certify",
      "decisionBy": {
        "givenName": "Foo",
        "id": "managed/user/1",
        "mail": "foo@test.com",
        "sn": "Test",
        "userName": "FooTest",
      },
      "decisionDate": "2023-02-28T15:12:25+00:00",
      "displayName": "Franklin Horne",
      "givenName": "Franklin",
      "id": "test",
      "linkQualifier": "default",
      "mail": "franklin.horne@usercreator.gen",
      "mailNickname": "fhorne",
      "manager": null,
      "metadata": {},
      "otherMails": [],
      "proxyAddresses": [],
      "surname": "Horne",
      "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      "userType": "Member",
    },
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "rightAlign" with v-bind value \`false\` will render rightAlign="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BMedia",
    "class="align-items-center"",
    "no-body=""",
    ">",
    "
",
    " at <AccountDetailsTab",
    "account=",
    {
      "account": {
        "displayName": "Franklin Horne",
        "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      },
      "accountEnabled": true,
      "accountType": "type",
      "decision": "certify",
      "decisionBy": {
        "givenName": "Foo",
        "id": "managed/user/1",
        "mail": "foo@test.com",
        "sn": "Test",
        "userName": "FooTest",
      },
      "decisionDate": "2023-02-28T15:12:25+00:00",
      "displayName": "Franklin Horne",
      "givenName": "Franklin",
      "id": "test",
      "linkQualifier": "default",
      "mail": "franklin.horne@usercreator.gen",
      "mailNickname": "fhorne",
      "manager": null,
      "metadata": {},
      "otherMails": [],
      "proxyAddresses": [],
      "surname": "Horne",
      "userPrincipalName": "fhorne@jstilton1973unfinishedlife.onmicrosoft.com",
      "userType": "Member",
    },
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
]
`;
